youtube_dl/extractor/googledrive.py

```diff
@@ -53,6 +64,98 @@ def _extract_url(webpage):
+                    'v': video_id,
+                })
+        except ExtractorError as ee:
+            self.report_warning('unable to download video subtitles: %s' % error_to_compat_str(ee))
+        if self._CAPTIONS_BY_COUNTRY_XML is not None:
+            caption_available_extensions = self._CAPTIONS_BY_COUNTRY_XML.findall('format')
+            for caption_extension in caption_available_extensions:
+                if caption_extension.attrib.get('fmt_code') and not caption_extension.attrib.get('default'):
+                    self._CAPTION_FORMATS_EXT.append(caption_extension.attrib['fmt_code'])
+
+    def _get_subtitles(self, video_id, video_subtitles_id, hl):
+        if not video_subtitles_id or not hl:
+            return None
+        if self._CAPTIONS_BY_COUNTRY_XML is None:
+            self._set_captions_data(video_id, video_subtitles_id, hl)
+            if self._CAPTIONS_BY_COUNTRY_XML is None:
+                return None
+
+        subtitles = {}
+        subtitle_available_tracks = self._CAPTIONS_BY_COUNTRY_XML.findall('track')
+        for subtitle_track in subtitle_available_tracks:
+            if not subtitle_track.attrib.get('lang_code'):
+                continue
+            subtitle_lang_code = subtitle_track.attrib['lang_code']
+            subtitle_format_data = []
+            for subtitle_format in self._CAPTION_FORMATS_EXT:
+                query = {
+                    'vid': video_subtitles_id,
+                    'v': video_id,
+                    'lang': subtitle_lang_code,
+                    'fmt': subtitle_format,
+                    'name': '',
+                    'kind': '',
+                }
+                subtitle_format_data.append({
+                    'url': update_url_query('https://drive.google.com/timedtext?vid=%s&v=%s&type=track&lang=%s&name&kind&fmt=%s', query),
+                    'ext': subtitle_format,
+                })
+            subtitles[subtitle_lang_code] = subtitle_format_data
+        if not subtitles:
+            self.report_warning('video doesn\'t have subtitles')
+        return subtitles
+
+    def _get_automatic_captions(self, video_id, video_subtitles_id, hl):
+        if not video_subtitles_id or not hl:
+            return None
+        if self._CAPTIONS_BY_COUNTRY_XML is None:
+            self._set_captions_data(video_id, video_subtitles_id, hl)
+            if self._CAPTIONS_BY_COUNTRY_XML is None:
+                return None
+        self.to_screen('%s: Looking for automatic captions' % video_id)
+
+        subtitle_original_track = self._CAPTIONS_BY_COUNTRY_XML.find('track')
+        if subtitle_original_track is None:
+            return None
+        if not subtitle_original_track.attrib.get('lang_code'):
+            return None
+        subtitle_original_lang_code = subtitle_original_track.attrib['lang_code']
+
+        automatic_captions = {}
+        automatic_caption_available_targets = self._CAPTIONS_BY_COUNTRY_XML.findall('target')
```

Comment:
Please clarify - the value of `subtitle_original_lang_code` will never change once it has been assigned, so what is the purpose of moving this into the `for` loop?

We need this value so that we can populate the query string for automatic captions later on in the method.